app/models/school.rb

```diff
@@ -44,7 +50,15 @@ def rejected?
   end
 
   def verify!
-    update!(verified_at: Time.zone.now)
+    attempts = 0
+    begin
+      update!(verified_at: Time.zone.now, code: SchoolCodeGenerator.generate)
+    rescue ActiveRecord::RecordInvalid => e
+      raise unless e.record.errors[:code].include?('has already been taken') && attempts <= 5
+
+      attempts += 1
+      retry
+    end
```

Comment:
Minor: I don't think you need the `begin` / `end` statements here, do you?

Reply:
I started without them but it meant that `attempts` was always reset to 0 after the call to `retry`. I considered setting an instance variable (e.g. `@verification_attempts`) in `after_initialize` but didn't think it bought enough to justify moving the counter away from where it's used.

Reply:
Doh! Of course - I completely missed that 👍